mdevaluate/correlation.py

```diff
@@ -77,31 +121,42 @@ def shifted_correlation(function, frames,
             skip = frames._slice.start / len(frames)
         except (TypeError, AttributeError):
             skip = 0
-    assert window + skip < 1
+    assert window + skip < 1  
 
     start_frames = np.unique(np.linspace(
         len(frames) * skip, len(frames) * (1 - window),
         num=segments, endpoint=False, dtype=int
     ))
-    num_frames = int(len(frames) * (window))
+    num_frames = int(len(frames) * (window)) 
 
     idx = index_distribution(0, num_frames)
 
+
     def correlate(start_frame):
         shifted_idx = idx + start_frame
         return correlation(function, map(frames.__getitem__, shifted_idx))
 
+
     result = 0 if average else []
-    for i, start_frame in enumerate(start_frames):
-        logger.debug('shifted_correlation: segment {}/{} (index={})'.format(i + 1, segments, start_frame))
+    count  = 0 if average else []
+    for i, start_frame in enumerate(start_frames): 
+        act_result, act_count = correlate(start_frame)
+        act_result = np.moveaxis(np.array(list(act_result)),0,-1)
         if average:
-            result += np.array(list(correlate(start_frame)))
+            result += act_result * act_count[...,np.newaxis]
+            count  += act_count
         else:
-            result.append(list(correlate(start_frame)))
-    result = np.array(result)
+            result.append(act_result)
+            count.append(act_count)
     if average:
-        result = result / len(start_frames)
+        np.divide(result, count[...,np.newaxis], out = result, where = count[...,np.newaxis] != 0)
+        count  = count  / len(start_frames)
+    else:
+        result = np.asarray(result)
```

Comment:
Sehe ich das richtig, wenn `average=False` ist, ist `result` noch mit den counts multipliziert?

Reply:
Wenn average = False, dann kommt der nicht in Zeile 146, sondern in 149 - müsste passen.